zipline/assets/_assets.pyx

```diff
@@ -298,6 +313,80 @@ cdef class Future(Asset):
         super_dict['contract_multiplier'] = self.contract_multiplier
         return super_dict
 
+"""
+TODO MB - rework this we need a Currency object and we need a FX pair
+
+ok we should use this a CurrencyPair, Currency could just be an enum?
+"""
+
+cdef class CurrencyPair(Asset):
+    cdef readonly object pair
+    cdef readonly object major
+    cdef readonly object minor
+    cdef readonly float cvf
+
+    def __cinit__(self,
+                  int sid,  # sid is required
+                  object symbol="",
+                  object pair="",
+                  object major="",
+                  object minor="",
+                  object start_date=None,
+                  float cvf=1,
+                  object asset_name="",
+                  object end_date=None,
+                  object notice_date=None,
+                  object first_traded=None,
```

Comment:
I use pycharm/intellij and I was getting a constructor error in the file coming up, I never checked this on the compiler side as I just left them in, happy to remove them.